sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -231,10 +231,15 @@ protected String visitBinaryArithmetic(String name, String l, String r) {
     return l + " " + name + " " + r;
   }
 
+  @Deprecated
   protected String visitCast(String l, DataType dataType) {
```

Comment:
let's remove this method directly. This is for users to override, it's worse if we keep it and user's override does nothing.